pkg/operation/botanist/constraints_check.go

```diff
@@ -0,0 +1,146 @@
+		return nil, fmt.Errorf("could not get MutatingWebhookConfigurations of Shoot cluster to check if Shoot can be hibernated")
+	}
+
+	for _, webhookConfig := range mutatingWebhookConfigs.Items {
+		if exitConstraint := checker.checkMutatingWebhookConfig(constraint, webhookConfig); exitConstraint != nil {
+			return exitConstraint, nil
+		}
+	}
+
+	c := gardencorev1alpha1helper.UpdatedCondition(constraint, gardencorev1alpha1.ConditionTrue, "NoProblematicWebhooks", "Shoot can be hibernated.")
+	return &c, nil
+}
+
+func (c *ConstraintsChecker) checkValidatingWebhookConfig(constraint gardencorev1alpha1.Condition, webhookConfig admissionregistrationv1beta1.ValidatingWebhookConfiguration) *gardencorev1alpha1.Condition {
+	return c.CheckWebhooks(constraint, "ValidatingWebhookConfiguration", webhookConfig.Name, webhookConfig.Webhooks)
+}
+
+func (c *ConstraintsChecker) checkMutatingWebhookConfig(constraint gardencorev1alpha1.Condition, webhookConfig admissionregistrationv1beta1.MutatingWebhookConfiguration) *gardencorev1alpha1.Condition {
+	return c.CheckWebhooks(constraint, "MutatingWebhookConfiguration", webhookConfig.Name, webhookConfig.Webhooks)
+}
+
+// CheckWebhooks checks all webhooks in the Shoot to determine if hibernation is not possible due to problematic Webhooks.
+// Problematic webhooks are: ValidationWebhookConfigurations and MutatingWebhookConfigurations with rules
+// for CREATE pods or nodes and failurePolicy=Fail and webhooks[].clientConfig.url == nil.
+// Otherwise, we can never wake up this shoot cluster again as new nodes cannot get created,
+// or our system component pods cannot get created (because the validating webhook pod is not yet running).
+func (c *ConstraintsChecker) CheckWebhooks(constraint gardencorev1alpha1.Condition, configKind string, configName string, webhooks []admissionregistrationv1beta1.Webhook) *gardencorev1alpha1.Condition {
+	for _, webhook := range webhooks {
+		if webhook.FailurePolicy != nil && *webhook.FailurePolicy != admissionregistrationv1beta1.Fail {
+			// in admissionregistration.k8s.io/v1 FailurePolicy is defaulted to `Fail`
+			// see https://github.com/kubernetes/api/blob/release-1.16/admissionregistration/v1/types.go#L195
+			// and https://github.com/kubernetes/api/blob/release-1.16/admissionregistration/v1/types.go#L324
+			// therefore, don't ignore FailurePolicy==nil
+			continue
+		}
+
+		if webhook.ClientConfig.URL != nil && len(*webhook.ClientConfig.URL) > 0 {
+			// Webhooks with external URL (out-of-cluster) are not problematic
+			continue
+		}
+
+		for _, rule := range webhook.Rules {
+			apiGroups := sets.NewString(rule.APIGroups...)
+			resources := sets.NewString(rule.Resources...)
+
+			if apiGroups.Has(corev1.GroupName) && resources.HasAny("pods", "nodes") {
+				for _, op := range rule.Operations {
+					if op == admissionregistrationv1beta1.Create || op == admissionregistrationv1beta1.OperationAll {
```

Comment:
We should also consider the `UPDATE` verb. If a `Node` can be created but not updated then the kubelet can never mark it as ready, i.e., no pods can be scheduled (unless inside host network).